enumerables.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # rubocop:disable Style/GuardClause, Style/RedundantSelf, Style/For
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    if block_given?
+      for i in 0...self.length
+        yield(self [i])
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      for i in 0...self.length
+        yield(self[i], i)
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    if block_given?
+      return_arr = []
+      self.my_each { |x| return_arr << x if yield(x) }
+      return_arr.nil? ? (return false) : (return return_arr)
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      self.my_select { |x| yield(x) }.length == self.length ? (return true) : (return false)
+    elsif !pattern.nil?
+      if pattern.is_a?(Regexp)
+        self.my_each { |x| return false unless pattern.match(x.to_s) }
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return false unless x.is_a?(pattern) }
+      else
+        self.my_each { |x| return false unless x == pattern }
+      end
+    else
+      self.my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given?
+      self.my_select { |x| yield(x) }.length.positive? ? (return true) : (return false)
+    elsif !pattern.nil?
+      if pattern.is_a?(Regexp)
+        self.my_each { |x| return true if pattern.match(x.to_s) }
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return true if x.is_a?(pattern) }
+      else
+        self.my_each { return true if x == pattern }
+      end
+    else
+      self.my_each { |x| return true if x }
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    if block_given?
+      self.my_select { |x| yield(x) }.length.positive? ? (return true) : (return false)
+    elsif !pattern.nil?
+      if pattern.is_a?(Regexp)
+        self.my_each { |x| return false if pattern.match(x.to_s) }
+      elsif pattern.is_a?(Class)
+        self.my_each { |x| return false if x.is_a?(pattern) }
+      else
+        self.my_each { |x| return false if x == pat }
+      end
+    else
+      self.my_each { |x| return false if x }
+    end
+    true
+  end
+
+  def my_count(elem = nil)
+    return self.length unless block_given?
+
+    if elem.nil?
+      return self.my_select { |x| yield(x) }.length
+    else
+      count = 0
+      self.my_each { |x| count += 1 if x == elem }
+      return count
+    end
+  end
```

Comment:
**[REQUIRED]**
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
     
       expected: 13
            got: 100
     
       (compared using ==)
```